hypothesisfuzzer/fuzzing/repo_fuzzer.py

```
@@ -122,18 +125,54 @@ def _clone_git(self, git_url):
         except Exception:
             logger.error('Unable to access repository %s.', self.name,
                          exc_info=True)
+
             return generic_error(msg="Error cloning/pulling Git Repo! " +
                                      "Please ensure you have access.")
 
     def _create_venv(self):
 
+        def pip_install(target):
+            # Target is a string
+
+            return subprocess.run(['venv/bin/pip', 'install'] + target,
+                                  cwd=self.name)
+
         logger.debug('Creating virtual environment for repository %s.',
                      self.name)
 
-        virtualenv.create_environment(self.name + '/venv')
-        subprocess.call([self.name + '/venv/bin/pip',
-                        'install', '-r', self.name + '/requirements.txt'])
+        if not os.path.isdir(self.name + '/venv'):
+            virtualenv.create_environment(self.name + '/venv')
+
+        if "dependencies" in self.config:
+            # Install dependencies
+
+            for dep_name, target in self.config["dependencies"].items():
+                if os.path.isfile(self.name + '/' + target):
+                    to_install = "-r " + target
+                else:
+                    to_install = target
+
+                install_result = pip_install(to_install)
 
+                if install_result.returncode != 0:
+                    self._ready = False
+                    msg = "Failed to install dependency set " \
+                        + dep_name + ': ' + target
+                    self._status = msg
+                    logger.error(msg)
+
+                    return
```

Do you want to return an exception or just ignore this?